### PR TITLE
Removal of leaves balance

### PR DESCRIPTION
**Pull request by AbdoHamed-TheBadNomads**

### CODE REVIEW

File: Leave/leave_db.py [hunk 1]
```diff
@@ -29,12 +30,17 @@ def GetEmergencyLeavesForYear(member_id, year):
     return leaves
 
 def GetLeaveBalance(member_id, leave_type):
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_db.py:32]: Let's change this to `GetAnnualLeaveBalance` cause we shouldn't need to check for balance for other leave types

File: Leave/leave_db.py [hunk 2]
```diff
@@ -29,12 +30,17 @@ def GetEmergencyLeavesForYear(member_id, year):
...
-    db.GetDBCursor().execute(f"SELECT balance FROM [leavesBalance] WHERE member_id = {member_id} AND leave_type = '{leave_type}'")
-    leaves_balance = [dict(zip([column[0] for column in db.GetDBCursor().description], row)) for row in db.GetDBCursor().fetchall()][0]
-    return float(leaves_balance["balance"])
-
-def GetLeaveTypesWithBalance():
-    db.GetDBCursor().execute('SELECT * FROM [leaveTypesWithBalance]')
+    start_date = member_db.GetMemberByID(member_id)["start_date"]
+    initial_balance = utils.CalculateProrataForLeave(leave_type, start_date)
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_db.py:34]: `CalculateProratedAnnualLeave`

File: SQL_Setup_3.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+DROP COLUMN starting_balance;
+
+DELETE FROM leavesBalance WHERE leave_type = 'Sick';
+
+DROP TABLE leavesBalance
+
+CREATE TABLE extraBalance (
+    id int NOT NULL IDENTITY(1,1) PRIMARY KEY,
+    [creditor_id] bigint NOT NULL FOREIGN KEY REFERENCES members(id),
+    [receiver_id] bigint NOT NULL,
```
Comment (AntWong-TheBadNomads) [inline SQL_Setup_3.sql:13]:
```suggestion
    [receipient_id] bigint NOT NULL,
```

File: Leave/leave_db.py (hunk 1 above)
Comment (AbdoHamed-TheBadNomads) [reply Leave/leave_db.py]: Done

File: Leave/leave_db.py (hunk 2 above)
Comment (AbdoHamed-TheBadNomads) [reply Leave/leave_db.py]: Done

File: SQL_Setup_3.sql (hunk 3 above)
Comment (AbdoHamed-TheBadNomads) [reply SQL_Setup_3.sql:13]: Done

File: SQL_Setup_3.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+sp_rename 'leaveTypesWithBalance', 'leaveTypes'
+
+ALTER TABLE leaveTypes
+DROP COLUMN starting_balance;
+
+DELETE FROM leavesBalance WHERE leave_type = 'Sick';
```
Comment (AntWong-TheBadNomads) [inline SQL_Setup_3.sql:6]: redundant

Comment (AbdoHamed-TheBadNomads) [reply SQL_Setup_3.sql]: Done

**Pull request merged**